s3fs/tests/test_s3fs.py

```diff
@@ -113,12 +132,19 @@ def test_rm(s3):
     assert s3.exists(a)
     s3.rm(a)
     assert not s3.exists(a)
+    with pytest.raises((OSError, IOError)):
+        s3.rm(test_bucket_name+'/nonexistent')
+    with pytest.raises((OSError, IOError)):
+        s3.rm('nonexistent')
+    s3.rm(test_bucket_name+'/nested', recursive=True)
+    assert not s3.exists(test_bucket_name+'/nested/nested2/file1')
 
 
 def test_anonymous_access():
     with ignoring(NoCredentialsError):
-        fs = S3FileSystem(anon=True)
-        fs.ls(test_bucket_name)
+        s3 = S3FileSystem(anon=True)
+        assert s3.ls('') == []
+        ## TODO: public bucket doesn't work through moto
```

Comment:
ACL did not seem to fix this. I'll look into it, but intend to leave as-is for now.